WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteViewModel.kt

```diff
@@ -58,6 +63,15 @@ class MySiteViewModel @Inject constructor(
         } else {
             listOf()
         }
+        UiModel(currentAvatarUrl.orEmpty(), items)
+    }
+
+    init {
+        updateAccountAvatarUrl()
```

Comment:
I'm wondering if it could ever happen that the user doesn't have an avatar (so the URL is empty) and then selects a new avatar but this field never gets updated. In the previous version the avatar URL is updated in `onCreateOptionsMenu`. Maybe it would make sense to update it in `onResume` if it has changed?

Reply:
Good question! I actually tested this scenario and thought it wasn't necessary as the avatar was being correctly updated after coming back from the Me screen, but your question made me realize that this was just a side effect of the call we are already making to `MySiteViewModel::updateSite` from the fragment's `onResume`. 

I've now moved the method to refresh the avatar url out of the ViewModel's `init` block so it can be called from the fragment's `onResume`, to make the intended behavior explicit.

Reply:
thanks, I've tested it as well and it works because the logic is strange. The avatar URL is always the same but it shows a different image in the Gravatar loader. I can't say I really understand why but I assume that the backend just replaces the image on the same URL. However, I think there could be a case where the avatar URL is empty and if that happens, I think we'd never recover. Thanks for the change!